pkg/analysis/passes/legacyplatform/legacyplatform.go

```diff
@@ -5,34 +5,59 @@ import (
 
 	"github.com/grafana/plugin-validator/pkg/analysis"
 	"github.com/grafana/plugin-validator/pkg/analysis/passes/modulejs"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/published"
 )
 
 var (
-	legacyPlatform = &analysis.Rule{Name: "legacy-platform", Severity: analysis.Warning}
+	legacyPlatform = &analysis.Rule{Name: "legacy-platform", Severity: analysis.Error}
 )
 
 var Analyzer = &analysis.Analyzer{
 	Name:     "legacyplatform",
-	Requires: []*analysis.Analyzer{modulejs.Analyzer},
+	Requires: []*analysis.Analyzer{modulejs.Analyzer, published.Analyzer},
 	Run:      run,
 	Rules:    []*analysis.Rule{legacyPlatform},
 }
 
+var legacyDetectionRegexes = []*regexp.Regexp{
+	// regexp.MustCompile(`['"](app/core/.*?)|(app/plugins/.*?)['"]`),
+	regexp.MustCompile(`['"](app/core/utils/promiseToDigest)|(app/plugins/.*?)|(app/core/core_module)['"]`),
+	regexp.MustCompile(`from\s+['"]grafana\/app\/`),
+	regexp.MustCompile(`System\.register\(`),
+}
+
 func run(pass *analysis.Pass) (interface{}, error) {
-	module := pass.ResultOf[modulejs.Analyzer].([]byte)
-
-	var (
-		reactExp   = regexp.MustCompile(`(@grafana/data)`)
-		angularExp = regexp.MustCompile(`([\s"']grafana/app/)`)
-	)
-
-	if angularExp.Match(module) && !reactExp.Match(module) {
-		pass.ReportResult(pass.AnalyzerName, legacyPlatform, "module.js: uses legacy plugin platform", "The plugin uses the legacy plugin platform (angularjs). Please migrate the plugin to use the new plugins platform.")
-	} else {
-		if legacyPlatform.ReportAll {
-			legacyPlatform.Severity = analysis.OK
-			pass.ReportResult(pass.AnalyzerName, legacyPlatform, "module.js: uses current plugin platform", "")
+
+	_, ok := pass.ResultOf[published.Analyzer].(*published.PluginStatus)
+
+	// we don't fail published plugins for using angular
+	if ok {
+		legacyPlatform.Severity = analysis.Warning
+	}
+
+	moduleJsMap, ok := pass.ResultOf[modulejs.Analyzer].(*map[string][]byte)
+	if !ok || moduleJsMap == nil {
```

Comment:
Nit: I would change the moduleJsMap check to:

```suggestion
	if !ok || len(moduleJsMap) == 0 {
```

len of a nil map is 0, and so is an empty map (map[...]...{}), but an empty map is not equal to nil:
https://go.dev/play/p/f5njCFt3Veb